updater/lib/dependabot/updater/operations/create_group_security_update_pull_request.rb

```diff
@@ -81,6 +63,42 @@ def perform
                     :dependency_snapshot,
                     :error_handler,
                     :created_pull_requests
+
+        def group
+          return @group if defined?(@group)
+
+          # make a temporary fake group to use the existing logic
+          @group = Dependabot::DependencyGroup.new(
+            name: "#{job.package_manager} at #{job.source.directory || '/'} security update",
+            rules: {
+              "patterns" => "*" # The grouping is more dictated by the dependencies passed in.
+            }
+          )
+          dependency_snapshot.job_dependencies.each do |dep|
+            @group.dependencies << dep
+          end
+          @group
+        end
+
+        def dependency_change
+          return @dependency_change if defined?(@dependency_change)
+
+          if job.source.directories.nil?
+            @dependency_change = compile_all_dependency_changes_for(group)
+          else
+            dependency_changes = job.source.directories.map do |directory|
+              job.source.directory = directory
+              # Fixes not updating because it already updated in a previous group
+              dependency_snapshot.handled_dependencies.clear
+              compile_all_dependency_changes_for(group)
+            end
```

Comment:
This is a cool approach! I think it makes a lot of sense. I do think we should make the logic of swapping the job directory more explicit though. I'll take a stab at refactoring it

Reply:
I hear you, but I wouldn't worry too much about refactoring and polishing yet. We can do that in follow-on PRs. For now I think it's important to get this out and tested so we can make progress!